packages/commonwealth/client/scripts/views/pages/CommunityManagement/Integrations/SpamLevel/SpamLevel.tsx

```diff
@@ -0,0 +1,124 @@
+import useRunOnceOnCondition from 'client/scripts/hooks/useRunOnceOnCondition';
+import { buildUpdateCommunityInput } from 'client/scripts/state/api/communities/updateCommunity';
+import { CWToggle } from 'client/scripts/views/components/component_kit/cw_toggle';
+import { CWSelectList } from 'client/scripts/views/components/component_kit/new_designs/CWSelectList';
+import { notifyError, notifySuccess } from 'controllers/app/notifications';
+import React, { useCallback, useState } from 'react';
+import app from 'state';
+import {
+  useGetCommunityByIdQuery,
+  useUpdateCommunityMutation,
+} from 'state/api/communities';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
+import './SpamLevel.scss';
+
+const SpamLevel = () => {
+  const communityId = app.activeChainId() || '';
+  const { data: community, isLoading: isLoadingCommunity } =
+    useGetCommunityByIdQuery({
+      id: communityId,
+      enabled: !!communityId,
+      includeNodeInfo: false,
+    });
+
+  const { mutateAsync: updateCommunity } = useUpdateCommunityMutation({
+    communityId: community?.id || '',
+  });
+
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+  const [spamTierLevel, setSpamTierLevel] = useState(-1);
+
+  useRunOnceOnCondition({
+    callback: () => {
+      const tier =
+        typeof community?.spam_tier_level === 'number'
+          ? community?.spam_tier_level
+          : -1;
+      setIsEnabled(tier >= 0);
+      setSpamTierLevel(tier >= -1 && tier <= 2 ? tier : -1);
```

Comment:
i would prefer to use enum instead od -1, 0, 1 numbers

Reply:
using enums in client, we can move to shared in a later PR